lib/internal/readline/interface.js

```diff
@@ -928,6 +954,29 @@ class Interface extends InterfaceConstructor {
     this[kRefreshLine]();
   }
 
+  [kMoveDownOrHistoryNext]() {
+    const { cols, rows } = this.getCursorPos();
+    const splitLine = StringPrototypeSplit(this.line, '\n');
+    if (!this.historyIndex && rows === splitLine.length) {
+      return;
+    }
+    // Go to the next history only if the cursor is in the first line of the multiline input.
+    // Otherwise treat the "arrow down" as a movement to the next row.
+    if (splitLine.length > 1 && rows < splitLine.length - 1) {
+      const currentLine = splitLine[rows];
+      const nextLine = splitLine[rows + 1];
+      const amountToMove = (cols > nextLine.length + 1) ?
+        currentLine.length - cols + nextLine.length + 3 :
+        currentLine.length + 1;
```

Comment:
The plus and minus parts are difficult to follow. Could you explain those please? :)